pkg/model/lattice/iamauthpolicy.go

```diff
@@ -0,0 +1,60 @@
+package lattice
+
+import (
+	"fmt"
+
+	anv1alpha1 "github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils"
+	"github.com/aws/aws-sdk-go/aws/arn"
+)
+
+type AuthPolicyKind string
+
+const (
+	GatewayAuthPolicy AuthPolicyKind = "gateway"
+	ServiceAuthPolicy AuthPolicyKind = "service"
+)
+
+type IAMAuthPolicy struct {
+	Kind         AuthPolicyKind
+	ResourceName string
+	Policy       string
+}
+
+type IAMAuthPolicyResource struct {
+	ARN  arn.ARN
+	Id   string
```

Comment:
I remember in the vpc lattice, AuthPolicy don't have ARN and ID?

Reply:
probably better to remove arn and leave only ID, seems confusing having different identifiers